### PR TITLE
Issue 337: Remove compact() operation

### PR DESCRIPTION
**Pull request by osopardo1**
## Description

Changes #337

## Type of change

Describe the change you're making: how it affects the API, user experience...

New feature.

## Checklist:

Here is the list of things you should do before submitting this pull request:

- [x] New feature / bug fix has been committed following the [Contribution guide](https://github.com/Qbeast-io/qbeast-spark/blob/main/CONTRIBUTING.md).
- [x] Add logging to the code following the [Contribution guide](https://github.com/Qbeast-io/qbeast-spark/blob/main/CONTRIBUTING.md).
- [x] Add comments to the code (make it easier for the community!).
- [ ] Change the documentation.
- [x] Add tests.
- [x] Your branch is updated to the main branch (dependent changes have been merged).

## How Has This Been Tested? (Optional)

No tests are added. Those involving the compact() operation are removed.

### CODE REVIEW

**Comment by Jiaweihu08** [comment]
Also, simplify the `QbeastTable.optimize` methods, as there are a few combinations of input arguments with default empty options.

**Comment by osopardo1** [comment]
> Also, simplify the `QbeastTable.optimize` methods, as there are a few combinations of input arguments with default empty options.

working on it

File: src/main/scala/io/qbeast/spark/QbeastTable.scala [hunk 1]
```diff
@@ -83,10 +82,12 @@ class QbeastTable private (
...
   }
 
-  def optimize(options: Map[String, String] = Map.empty): Unit = {
-    if (!isStaging(latestRevisionAvailableID)) {
-      optimize(latestRevisionAvailableID, options)
-    }
+  def optimize(options: Map[String, String]): Unit = {
+    optimize(latestRevisionAvailableID, options)
+  }
+
```
Comment (Jiaweihu08) [inline src/main/scala/io/qbeast/spark/QbeastTable.scala:88]: We should add a method when the user wants to optimize a RevisionID with not options:
```scala
def optimize(revisionID: RevisionID): Unit = {
  optimize(revisionID, Map.empty[String, String])
}
```

Then, another method for a sequence of input file paths without options:
```scala
def optimize(files: Seq[String]): Unit =
  optimize(files, Map.empty[String, String])
```

File: src/main/scala/io/qbeast/core/model/DataWriter.scala [hunk 2]
```diff
@@ -63,7 +63,7 @@ trait DataWriter[DATA, DataSchema, FileDescriptor] {
    * @return
    *   the sequence of files written and deleted
    */
-  def compact(
+  def optimize(
```
Comment (Jiaweihu08) [inline src/main/scala/io/qbeast/core/model/DataWriter.scala:66]: Please also update the method description

Comment (Jiaweihu08) [reply src/main/scala/io/qbeast/core/model/DataWriter.scala:66]: For example, "Reindex the selected files to bring the data layout closer to the dictates of the index."

File: src/main/scala/io/qbeast/spark/QbeastTable.scala (hunk 1 above)
Comment (osopardo1) [reply src/main/scala/io/qbeast/spark/QbeastTable.scala:92]: done

File: src/main/scala/io/qbeast/core/model/DataWriter.scala (hunk 2 above)
Comment (Jiaweihu08) [reply src/main/scala/io/qbeast/core/model/DataWriter.scala:66]: Do we have documentation explaining what optimization means now?

Comment (Jiaweihu08) [reply src/main/scala/io/qbeast/core/model/DataWriter.scala:66]: To be found in `/docs/QbeastFormat.md`

**Review by Jiaweihu08 — APPROVED** [review]

**Pull request merged**